### ISSUE TITLE
Dark Mode Toggle Not Working in 0.6.6 - Regression from 0.6.5

### ISSUE THREAD

**Issue by ebonura-fastly** [reflex-dev/reflex#4464]
**Describe the bug**
The dark mode toggle functionality from the official Reflex documentation breaks in version 0.6.6. The same example works correctly in version 0.6.5.

**To Reproduce**
1. Use the example code from https://reflex.dev/docs/recipes/others/dark-mode-toggle/
2. Run the code with Reflex 0.6.6
3. Attempt to toggle dark mode
4. Toggle fails to function as expected

Code/Link to Repo: Using official documentation example from https://reflex.dev/docs/recipes/others/dark-mode-toggle/

**Expected behavior**
Dark mode toggle should switch between light and dark themes as demonstrated in the documentation and as functioning in Reflex 0.6.5.

**Specifics**
- Python Version: 3.12
- Reflex Version: 0.6.6 (broken), 0.6.5 (working)
- OS: MacOS
- Browser: Tested in Chrome and Safari

**Additional context**
The issue appears to be a regression as the official documentation example functions correctly in version 0.6.5 but breaks in 0.6.6. The bug is reproducible across different browsers.

**Comment by Lendemor** [comment]
Tested locally.
I can reproduce the problem.

After further checking, it seems like the theme value is set correctly (see `DevTools` > `Application` > `Local Storage` > `theme`), but is somehow ignored.

**Comment by Lendemor** [comment]
@ebonura-fastly 
Hum, I've been doing more testing but it's not happening anymore despite not changing anything in the code.
Can you try running `reflex init` again maybe on your project and tell me if it keep happening?

**Comment by Lendemor** [comment]
After more testing, it seems like a page refresh seems to trigger the bug consistently.
1. Open the page
2. Go to light mode (when it's working)
3. Do a page reload.
4. Page is in dark mode even though all the variable are set to light.

Once the color mode is broken, it stay broken even with refresh.

However, if you do:
1. Switch back to color_mode dark.
2. Refresh the page.
3. Switch to color_mode light.
4. The page is in light mode again.

**Comment by Lendemor** [comment]
I added a page reload in the test above to try and catch the issue, but it doesn't, at least locally.

I suspect some browser specific issue.

**Comment by Lendemor** [comment]
While inspecting the page, I've noticed that the attributes `dark-native-active` show up in the <html> tag of the page, which seems to supersede the actual color mode management in reflex.

@ebonura-fastly do you also see this attributes ?

**Comment by Seikened** [comment]
<img width="454" alt="Captura de pantalla 2024-12-05 a la(s) 12 26 30 a m" src="https://github.com/user-attachments/assets/24b2d3e6-6129-45ba-8361-5e7c32ec5a15">

*Do you mean of this?*

**Comment by ebonura-fastly** [comment]
Thanks for checking this, @Lendemor . When switching between dark/light modes, the HTML tag only switches between:

```html
class="light" style="color-scheme: light"
```
or
```html
class="dark" style="color-scheme: dark"
```

I've tested this across multiple versions (0.6.5, 0.6.6, and 0.6.6.post2) on both Chrome and Safari, and don't see the `dark-native-active` attribute you mentioned.

I've also tried running `reflex init` and refreshing the page multiple times, but the issue persists

**Comment by Lendemor** [comment]
This show up for me if I refresh while in light mode.
![image](https://github.com/user-attachments/assets/d96bcb86-e6cc-49ac-b56c-6bde1a811b84)
Which seems to override the page and make it render in dark mode despite the other attribute being set to light.

This is the only case where I reproduced the issue.

**Comment by ebonura-fastly** [comment]
@Lendemor I've double-checked the issue with Chrome incognito and Safari (no extensions installed) - it persists in both. While I do use Dark Reader, the problem seems to occur even with all extensions disabled.

**Comment by Lendemor** [comment]
![Screen Recording 2024-12-05 at 14 57 58](https://github.com/user-attachments/assets/053de9e4-bbda-43b1-a28d-c7d6da03f788)

I'm running out of idea on what to test / how to find the problem.

This is the app I'm using to test : 

```python
import reflex as rx
from reflex.style import set_color_mode, color_mode, resolved_color_mode

app = rx.App()


@app.add_page
def index():
    return (
        rx.segmented_control.root(
            rx.segmented_control.item(
                rx.icon(tag="monitor", size=20),
                value="system",
            ),
            rx.segmented_control.item(
                rx.icon(tag="sun", size=20),
                value="light",
            ),
            rx.segmented_control.item(
                rx.icon(tag="moon", size=20),
                value="dark",
            ),
            on_change=set_color_mode,
            variant="classic",
            radius="large",
            value=color_mode,
        ),
        rx.text(rx.constants.Reflex.VERSION),
        # rx.color_mode.button(allow_system=True),
        rx.text(color_mode, id="current_color_mode"),
        rx.text(resolved_color_mode, id="resolved_color_mode"),
        rx.text(rx.color_mode_cond("LightMode", "DarkMode"), id="color_mode_cond"),
    )

```

**Comment by ebonura-fastly** [comment]
Hey @Lendemor I think I found the root cause, it is related to the appearance setting in rx.theme, which in 0.6.6 will override whatever is set in the segmented control, this was not happening in previous versions.

The reason why I couldn't switch is because I had the appearance set to light in my rx.theme to provide a default.

Thanks for your help and patience while we debugged this! I think the issue can be closed, I'll leave it to you and your team if this is the expected behavior, I do believe it should be noted in https://reflex.dev/docs/recipes/others/dark-mode-toggle/ though, just to avoid confusion as people may not have it working as expected

For reference, here's a minimal example showing the difference:

```python
import reflex as rx
from reflex.style import set_color_mode, color_mode, resolved_color_mode

def example_theme(appearance: str = None):
    return rx.theme(
        appearance=appearance
    )

def dark_mode_toggle():
    return rx.segmented_control.root(
        rx.segmented_control.item(rx.icon(tag="monitor", size=20), value="system"),
        rx.segmented_control.item(rx.icon(tag="sun", size=20), value="light"),
        rx.segmented_control.item(rx.icon(tag="moon", size=20), value="dark"),
        on_change=set_color_mode,
        variant="classic",
        radius="large",
        value=color_mode,
    )

def index():
    return rx.vstack(
        rx.heading("Dark Mode Demo"),
        dark_mode_toggle(),
        rx.text("Current mode: ", color_mode),
        rx.text("Resolved mode: ", resolved_color_mode),
        rx.text(rx.color_mode_cond("Light Theme", "Dark Theme")),
    )

# Comment/uncomment to see difference
# app = rx.App(theme=example_theme("light"))
app = rx.App(theme=example_theme())

app.add_page(index)
```

**Comment by Lendemor** [comment]
Hi @ebonura-fastly, thanks for figuring that out.

Imo it's definitely a regression, the `appearance` should only be the default for how the page is supposed to be when first opened, not a fixed value.

I'm not sure how to fix that bug (yet) but I'll update the test to take this use-case into account.

**Comment by Lendemor** [comment]
Found a fix for this, added it in #4467
